### ISSUE TITLE
Allowed memory exhausted on large pages with PATCH request from Android app

### ISSUE THREAD

**Issue by nanawel** [wallabag/wallabag#7679]
### Environment

* **Version**: 2.6.9 (latest on Docker at the time of writing)
* **Installation**: Docker
* **OS**: Debian (as host)
* **Database**: mariadb (latest)

### What steps will reproduce the bug?

From a **desktop Firefox**, bookmark a large page with images, for example <https://gonzai.com/shyle-zalewski-fait-il-toujours-autant-nimporte-quoi/> using Wallabagger with "Retrieve content from the browser by default when saving page" checked (I don't know if it makes a difference).

Then access and read the page **from you Android app**, and **archive it** ("mark as read").

A `PATCH` request is issued to the server, but with the default `memory_limit` from `php.ini` (128M if I'm correct) it will likely fail. If you have a reverse-proxy in front of your server, you should see a `499` HTTP code in return, but nothing is logged anywhere. That's the first issue because nothing in the Android app tells you that an error occured, you just won't receive any updates anymore and new items will no longer appear in the list.

Increase the `memory_limit` and restart the server until you find a value that allows the request to complete successfully. For the page given as an example, I originally had 256M but that wasn't enough, so I tested 512M, then 1G and finally 2G!

How is it possible to require so much memory just to "flip a bit" of the "is_archived" field? (okay I'm exaggerating I know, but you know what I mean)

There must be a better way to handle the `PATCH` request in that case, or is it because of what the Android app is sending?

**Comment by j0k3r** [comment]
Well I don't know how the android perform that "mark as read" for a given entry.

If instead of only _flipping_ a boolean it also sends the whole entry (ie: content, title, other boolean, etc..) it might be the reason why it tooks so long to handle.

**Comment by nanawel** [comment]
> If instead of only _flipping_ a boolean it also sends the whole entry (ie: content, title, other boolean, etc..) it might be the reason why it tooks so long to handle.

Yes this is what I fear. But I don't understand how this memory issue is never triggered using the Firefox extension nor the webapp. When bookmarking a huge page, I supposed the request and the processing was at least as big as what the Android app does when marking as read.
